### PR TITLE
EN-1562 [Java SDK] Refactor SDK to not use Request/Response objects

### PR DESCRIPTION
**Pull request by robert-par**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- https://incountry.atlassian.net/browse/EN-1562

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [x] Please put the target release vehicle this needs to be bundled with.
- [x] Did you increment the version? What is the new version?

### CODE REVIEW

**Comment by avdyakovs** [comment]
@robert-par fix sonar please

File: src/main/java/com/incountry/Storage.java [hunk 1]
```diff
@@ -153,84 +178,98 @@ public Record read(String country, String recordKey) throws StorageException, IO
...
         FindFilter findFilter = new FindFilter();
         findFilter.setVersionParam(new FilterStringParam(secretKeyCurrentVersion.toString(), true));
         BatchRecord batchRecord = find(country, findFilter,  new FindOptions(limit, 0));
         batchWrite(country, batchRecord.getRecords());
+        MigrateResult migrateResult = new MigrateResult();
+        migrateResult.setMigrated(batchRecord.getCount());
+        migrateResult.setTotalLeft(batchRecord.getTotal() - batchRecord.getCount());
 
-        return new MigrateResult(batchRecord.getCount(), batchRecord.getTotal() - batchRecord.getCount());
+        return migrateResult;
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/Storage.java]: why not use constructor for it?

File: src/main/java/com/incountry/Storage.java [hunk 2]
```diff
@@ -153,84 +178,98 @@ public Record read(String country, String recordKey) throws StorageException, IO
...
-        httpAgent.request(url, "POST", "{ \"records\" : " + new Gson().toJson(recordsStrings) + "}", false);
+        String url = getEndpoint(country, "/v2/storage/records/"  + country + "/batchWrite");
+        try {
+            httpAgent.request(url, "POST", "{ \"records\" : " + new Gson().toJson(recordsStrings) + "}", false);
+        } catch (IOException e) {
+            throw new StorageServerException(SERVER_ERROR_MESSAGE, e);
+        }
 
-        return true;
+        return new BatchRecord(new ArrayList<>(), 0, 0, 0, 0);
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/Storage.java]: it should return the created records, not empty BatchRecord instance :)

File: src/main/java/com/incountry/Storage.java [hunk 3]
```diff
@@ -153,84 +178,98 @@ public Record read(String country, String recordKey) throws StorageException, IO
...
+     * Find records in remote storage
+     * @param country country identifier
+     * @param filter object representing find filters
+     * @param options find options
+     * @return BatchRecord object which contains required records
+     * @throws StorageServerException if server connection failed or server response error
+     * @throws StorageCryptoException if decryption failed
+     */
+    public BatchRecord find(String country, FindFilter filter, FindOptions options) throws StorageServerException, StorageCryptoException {
+        if (country == null) throw new NullPointerException("Missing country");
```
Comment (RommeroInCountry) [inline src/main/java/com/incountry/Storage.java]: Once again - Let's either use IllegalArgumentException or StorageClientException. I'd personally go with the latter one.
And we need to change the error text to smth like "country/key cannot be null"

File: README.md [hunk 4]
```diff
@@ -261,10 +263,17 @@ Available request options for `FilterRangeParam`: `$lt`, `$lte`, `$gt`, `$gte`.
 If you need to find the first record matching filter, you can use the `findOne` method.
 It works the same way as `find` but returns the first record or `null` if no matching records.
 
+### Update one record
+If you need to update record, you can use `updateOne` method.
+```
+public Record updateOne(String country, FindFilter filter, Record record) throws StorageServerException, StorageCryptoException
+```
+It takes country code of the record, filters and new record object which must replace updating record.
+
```
Comment (RommeroInCountry) [inline README.md]: We'll be removing update one method soon. No need to add readme for it

**Pull request merged**